### PR TITLE
✨amp-autocomplete: Add "item" attribute to specify the relevant property of a datasource

### LINKED ISSUE #23814: Dynamic properties in autocomplete JSON response
## Describe the new feature or change to an existing feature you'd like to see

Currently `autocomplete` requires the `items` property on the datasource as hardcoded [here](https://github.com/ampproject/amphtml/blob/54a576f763f0816d9e24185fbe314780dfec771d/extensions/amp-autocomplete/0.1/amp-autocomplete.js#L269).

My suggestion is to add an option to select the name of the property.

An other suggestion is support for `object` instead of an array like `amp-list` with `items="."` and `single-item`.

Reason for this change is that not everyone can change the remote datasource.

### PR DESCRIPTION
**Pull request by caroqliu**
Closes #23814 

This change adds an attribute `items` to amp-autocomplete which defines the expression to locate the array to be rendered within the response. Updates documentation, unit tests, validations accordingly.

### CODE REVIEW

File: extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out [hunk 1]
```diff
@@ -132,6 +132,17 @@ FAIL
 |      <input>
 |    </amp-autocomplete>
 |
+|    <!-- Valid: amp-component with static data and item=property attribute -->
```
Comment (honeybadgerdontcare) [inline extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out]: s/item/items

File: extensions/amp-autocomplete/amp-autocomplete.md [hunk 2]
```diff
@@ -119,8 +119,8 @@ Example:
     </td>
   </tr>
   <tr>
-    <td width="40%"><strong>highlight-user-entry (optional)</strong></td>
```
Comment (honeybadgerdontcare) [inline extensions/amp-autocomplete/amp-autocomplete.md]: `highlight-user-entry` is still allowed in validation. Is it intended to remove this from documentation?

Comment (caroqliu) [reply extensions/amp-autocomplete/amp-autocomplete.md]: Not at all, thanks for catching this!

File: extensions/amp-autocomplete/amp-autocomplete.md [hunk 3]
```diff
@@ -123,6 +123,11 @@ Example:
     <td>If present, exposes the <code>autocomplete-partial</code> class on the substring within the suggested item that resulted in its match with the user input. This can be used to stylize the corresponding match to stand out to the user. <strong>Defaults to false.</strong>
     </td>
   </tr>
+  <tr>
+    <td width="40%"><strong>items (optional)</strong></td>
```
Comment (cathyxz) [inline extensions/amp-autocomplete/amp-autocomplete.md]: Nit: consider adding an example.

File: examples/autocomplete-states.example.json [hunk 4]
```diff
@@ -1,9 +1,63 @@
 {
```
Comment (wassgha) [inline examples/autocomplete-states.example.json]: nit: let's revert the formatting to what it used to be? probably accidental formatting?

File: extensions/amp-autocomplete/amp-autocomplete.md [hunk 5]
```diff
@@ -123,6 +123,11 @@ Example:
...
+    <td>Defines the expression to locate the array to be rendered within the response. This is a dot-notated expression that navigates via fields of the JSON response. The default value is <code>"items"</code>. The expected response: <code>{items: [...]}</code>.
```
Comment (wassgha) [inline extensions/amp-autocomplete/amp-autocomplete.md]: nit: to echo the language of the doc I'd probably say 

> Specifies the key to the data array within the JSON response. Nested keys can be expressed with a dot-notated value such as `field1.field2`. The default value is ...

**Pull request merged**